parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -218,9 +253,46 @@
             root_manifest.created,
             Pendulum.now(),
         )
-    return {
-        item[0]: item[1]
+    versions_list = [
+        (item[0], item[1])
         for item in sorted(
-            list(return_tree.items()), key=lambda item: (item[0][3], item[0][0], item[0][1])
+            list(return_tree.items()), key=lambda item: (item[0].late, item[0].id, item[0].version)
         )
-    }
+    ]
+    previous = None
+    new_list = []
+    # Merge duplicates with overlapping time frames as it can be caused by an update of a parent
+    # dir, also, if option is set, remove what can be guessed as empty manifests set before parent
+    # during sync
+    for item in versions_list:
+        if previous is not None:
+            # If same entry_id and version
+            if previous[0].id == item[0].id and previous[0].version == item[0].version:
```

Comment:
trop de `[0]`, il faut mieux avoir des variables intermédiaires qui aident à comprendre ce qu'on manipule

pourquoi ne pas remplacer `previous` par `previous_ts_bounded_entry` et `previous_manifest_data_and_path` ?
pareil pour `item` qu'on peut décomposer dès l'itérateur `for item_ts_bounded_entry, item_manifest_data_and_path in version_list`

(oui les noms son super long, faudrait peut être en trouver des mieux ^^)